File: platforms/swift/examples/core/renderer/load.swift
import FragmentColor

let renderer = Renderer()
let target = try await renderer.createTextureTarget([64, 64])

let mesh = Mesh()
try mesh.addVertex(
    try Vertex.pbr([0.0, 0.5, 0.0]).set("uv0", [0.5, 1.0])
)
// Raw 2×2 RGBA pixel bytes — uploaded lazily by """Renderer.load""" below.
// In practice the loader hands the setter encoded PNG/JPEG bytes (from a
// BIN chunk) or a file path (from a URI); the same """Into<TextureInput>"""
// vocabulary covers all of them.
let red_pixels = [
    255,   0,   0, 255,    0, 255,   0, 255,
      0,   0, 255, 255,  255, 255, 255, 255,
]
let red_tex = try await renderer.createTexture((red_pixels, [2, 2]))
let material = Material.pbr().baseColorTexture(red_tex)
let model = Model(mesh, material)
let scene = Scene()
try scene.add(model)

// Eager prewarm — uploads the pending texture(s) so the next render is
// GPU-only.
try await renderer.load(scene)
try renderer.render(scene, target)

File: platforms/python/examples/core/renderer/load.py
from fragmentcolor import Material, Mesh, Model, Renderer, Scene, Vertex

renderer = Renderer()
target = renderer.create_texture_target([64, 64])

mesh = Mesh()
mesh.add_vertex(
    Vertex([0.0, 0.5, 0.0]).set(Vertex.NORMAL, [0.0, 0.0, 1.0]).set(Vertex.UV0, [0.5, 1.0]).set(Vertex.COLOR0, [1.0, 1.0, 1.0, 1.0]).set(Vertex.UV1, [0.0, 0.0]),
)
# Raw 2×2 RGBA pixel bytes — uploaded lazily by `Renderer.load` below.
# In practice the loader hands the setter encoded PNG/JPEG bytes (from a
# BIN chunk) or a file path (from a URI); the same `Into<TextureInput>`
# vocabulary covers all of them.
red_pixels = [
    255,   0,   0, 255,    0, 255,   0, 255,
      0,   0, 255, 255,  255, 255, 255, 255,
]
material = Material.pbr().base_color_texture((red_pixels, [2, 2]))
model = Model(mesh, material)
scene = Scene()
scene.add(model)

# Eager prewarm — uploads the pending texture(s) so the next render is
# GPU-only.
renderer.load(scene)
renderer.render(scene, target)